build.gradle

```
@@ -0,0 +1,54 @@
+apply plugin: 'java'
+
+group = 'com.pi4j'
+version = '0.0.1'
+
+description = """Pi4J :: MINIMAL EXAMPLE :: Sample minimal project"""
+project.buildDir = 'target'
+
+tasks.withType(JavaCompile) {
+    options.encoding = 'UTF-8'
+}
+
+// Copy the Java modules in the distribution directory
+plugins.withType(JavaPlugin).configureEach {
+    java {
+        modularity.inferModulePath = true
+    }
+}
+task copyDependencies(type: Copy) {
+    from configurations.default
+    into 'target/distribution'
+}
+build.finalizedBy copyDependencies
+
+// Copy the generated application JAR in the distribution directory
+task copyJar(type: Copy) {
+    from jar
+    into 'target/distribution'
+}
+build.finalizedBy copyJar
+
+// Copy the run script in the distribution directory
+task copyRun(type: Copy) {
+    from 'assets/run.sh'
+    into 'target/distribution'
+}
+build.finalizedBy copyRun
```

This could be a single copy task:
```suggestion
tasks.register("copyDistribution", CopySpec) {
    from configurations.default
    from jar
    from 'assets/run.sh'
    into file("$buildDir/distribution")
}
tasks.named("build") {
    dependsOn("copyDistribution")
}
```

❗ Note that you could consider using the Gradle `application` plugin instead, see https://docs.gradle.org/current/userguide/application_plugin.html